src/renderable/sprite.js

```diff
@@ -269,28 +268,26 @@
             var ax = w * this.anchorPoint.x, ay = h * this.anchorPoint.y;
             renderer.translate(-ax, -ay);
             
-            if ((this.scaleFlag) || (angle !== 0)) {
-
+            if ((this.scaleFlag) || (this.angle !== 0) || (this._sourceAngle !== 0)) {
                 // translate to the defined anchor point
                 renderer.translate(xpos + ax, ypos + ay);
-                if (angle !== 0) {
-                    renderer.rotate(angle);
+                // rotate
+                if (this.angle !== 0) {
+                    renderer.rotate(this.angle);
                 }
                 // scale
                 if (this.scaleFlag) {
                     renderer.scale(this._scale.x, this._scale.y);
                 }
-
+                // remove image's TexturePacker/ShoeBox rotation
                 if (this._sourceAngle !== 0) {
-                    // swap w and h for rotated source images
+                    renderer.translate((this.height/2)-(xpos+ax), (this.width/2)-(ypos+ay));
+                    renderer.rotate(this._sourceAngle);
+                    xpos -= this.height/2;
+                    ypos -= this.width/2;
```

Comment:
I don't think the rotation is supposed to be performed around the center. Assume an image with a 2:1 or 1:2 ratio; rotating around the center will change the position of the image, relative to its anchor point.